File: Minimum-Path-Sum/Minimum-Path-Sum.cpp

class Solution {
public:
    int minPathSum(vector<vector<int> > &grid) {
        int n=grid.size();
        int m=grid[0].size();
        vector<int>dp(m+1,INT_MAX);
        dp[1]=0;
        for(int i=0;i<n;i++)
        {
            for(int j=0;j<m;j++)
            {
                dp[j+1]=min(dp[j],dp[j+1])+grid[i][j];
            }
        }
        return dp[m];
    }
};


 

File: solutions/Minimum-Path-Sum/Minimum-Path-Sum.js
/**
 * @param {number[][]} grid
 * @return {number}
 */
var minPathSum = function (grid) {
  const n = grid.length;
  const m = grid[0].length;
  const dp = Array.from({ length: m + 1 }, () => Number.MAX_SAFE_INTEGER);
  dp[1] = 0;
  for (let i = 0; i < n; i++) {
    for (let j = 0; j < m; j++) {
      dp[j + 1] = Math.min(dp[j], dp[j + 1]) + grid[i][j];
    }
  }
  return dp[m];
};